subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/msvcpp/DefaultVisualStudioLocator.java

```diff
@@ -105,105 +100,149 @@ public SearchResult locateDefaultVisualStudioInstall(File candidate) {
 
     private void initializeVisualStudioInstalls() {
         if (!initialised) {
-            locateInstallsInRegistry();
-            locateInstallInPath();
+            locateInstallsWith(commandLineLocator);
+
+            if (foundInstalls.size() < 1) {
+                locateInstallsWith(windowsRegistryLocator);
```

Comment:
i think we'd want to collect the union of those we find with `vswhere` and the registry (and maybe the path too), and select the best match from these.

Reply:
Well, we only support VS2010 and above, and `vswhere` will detect all of those versions, so it can detect a superset of what we can detect from the registry.  I don't think there is a need to use the registry if we get results from `vswhere`.  

Adding the path in really only seems to make sense when we start talking about modeling/specifying a version or other requirement (i.e. I want version X even if higher version Y is found).  With a specified version (or requirement), I can see us gathering up all the installations we can find (possibly including on the path), and then trying to figure out if any match the specification.  When we're just selecting the default (i.e. highest version), though, if we find something in `vswhere` or the registry, we're going to prefer that over some unknown version on the path, so there's no point in looking at the path unless we haven't found something elsewhere.

Right now, you can select a particular version by specifying an installation directory and then we try to figure out what version it is.  This is basically achieved by implementing a similar strategy - find all versions, look for one that matches the spec (i.e. one that has the specified installation directory).  The difference is that if we don't find the version through normal means (i.e. through `vswhere` or the registry), we still accept it with an "unknown" version.